tests/ui/base_test.py

```diff
@@ -30,3 +54,184 @@ def setContext(self):
+        EC.visibility_of_element_located(((UfOPageLayout.ADD_MANUALLY_FORM))))
+    name_paper_input = add_manually_form.find_element(
+        *UfOPageLayout.ADD_MANUALLY_INPUT_NAME)
+    name_input = name_paper_input.find_element(By.ID, 'input')
+    name_input.send_keys(BaseTest.TEST_USER_AS_DICT['name'])
+    email_paper_input = add_manually_form.find_element(
+        *UfOPageLayout.ADD_MANUALLY_INPUT_EMAIL)
+    email_input = email_paper_input.find_element(By.ID, 'input')
+    email_input.send_keys(BaseTest.TEST_USER_AS_DICT['email'])
+    submit_button = self.driver.find_element(
+        *UfOPageLayout.ADD_MANUALLY_SUBMIT_BUTTON)
+    submit_button.click()
+
+    # Wait for post to finish, can take a while.
+    WebDriverWait(self.driver, 20).until(
+        EC.invisibility_of_element_located(((
+            UfOPageLayout.ADD_MANUALLY_SPINNER))))
+
+  def remove_test_user(self, raiseException=True):
+    """Manually remove a test user using the landing page (the only way).
+
+    Args:
+      raiseException: True to raise an exception if the user is not found.
+    """
+    # Find the user and navigate to their details page.
+    self.driver.get(self.args.server_url + flask.url_for('landing'))
+    landing_page = LandingPage(self.driver)
+    user_list_item = landing_page.GetElement(LandingPage.USER_LIST_ITEM)
+    user_listbox = user_list_item.find_element(*LandingPage.GENERIC_LISTBOX)
+    user_item = self.find_item_in_listing(
+        user_listbox, BaseTest.TEST_USER_AS_DICT['name'])
+
+    if user_item is None:
+      if raiseException:
+        raise Exception
+      else:
+        return
+    else:
+      user_item.click()
+
+    # Click delete on that user.
+    details_modal = user_item.find_element(*LandingPage.DETAILS_MODAL)
+    WebDriverWait(self.driver, 10).until(
+        EC.visibility_of(details_modal))
+    delete_button = details_modal.find_element(*LandingPage.USER_DELETE_BUTTON)
+    delete_button.click()
+
+    # Wait for post to finish, can take a while.
+    WebDriverWait(self.driver, 20).until(
+        EC.invisibility_of_element_located(((
+            LandingPage.USER_DELETE_SPINNER))))
+
+  def find_item_in_listing(self, listing, name):
+    """Given the listing of items and a name, return the name's anchor.
+
+    Args:
+      listing: The paper-listbox element holding all items.
+      name: The name of an item to search for.
+
+    Returns:
+      The anchor element for visiting the given item's details page or None.
+    """
+    items = listing.find_elements(By.TAG_NAME, 'paper-icon-item')
+    for item in items:
+      # This can technically return multiple, but it will only return one.
+      div = item.find_elements(By.CLASS_NAME, 'first-div')[0]
+      strong = div.find_elements(By.TAG_NAME, 'strong')[0]
+      if name.lower() in strong.text.lower():
+        return item
+    return None
+
+  def add_test_server_helper(self, containingElement):
```

Comment:
Yes, the casing is confusing.  Here specifically, "containing_element" would be better & consistent.